test/utils/junit/junit.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package junit provides data structures to allow easy XML encoding
+// and decoding of JUnit test results.
```

Comment:
This xsd seems incomplete. It doesn't list half the fields I have defined (that I've seen in other sources including the ones ginkgo generates). From what I've gathered, there really is no consensus and it's mostly ad-hoc. It would be disingenuous to reference that xsd here. Given that, should it be trusting on required fields?